source/client/cl_vid.c

```diff
@@ -316,7 +317,11 @@ static bool VID_LoadRefresh( const char *name )
 	GetRefAPI_t GetRefAPI_f;
 
 	VID_UnloadRefresh();
+	
+	vid_ref_mempool = Q_CreatePool( NULL, "Refresh" );
+	struct mem_import_s memImport = DECLARE_MEM_STRUCT(vid_ref_mempool); 
 
+	import.memImport = &memImport;
```

Comment:
This becomes a dangling reference as soon as `VID_LoadRefresh` returns, so you might as well skip the dance of storing it in a local variable and returning a pointer to that, and just store the entire thing directly in `import` ?

Reply:
C doesn't have a dangling reference its a point to the struct that declares the import. I guess the second the function returns I the pointer become undefined. its only declared for this function so its probably not too big of an issue?

Reply:
I added an assertion and made the variable static so there is a dropped pointer